Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs

```diff
@@ -783,6 +645,87 @@ private string GetRandomString()
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Calculat casual leava and sick leave from the date of joining
+        /// </summary>
+        /// <param name="dateTime"></param>
+        /// <returns></returns>
+        private LeaveCalculator CalculateAllowedLeaves(DateTime dateTime)
```

Comment:
Private method CalculateAllowedLeaves is called by the public method "AddUserAsync" and I have already checked it in my local machine for the unit test and I found that it works.